packages/node_modules/@webex/webex-core/src/lib/services/service-host.js

```diff
@@ -0,0 +1,250 @@
+import Url from 'url';
+
+import {SERVICE_CATALOGS} from './constants';
+
+/**
+ * ServiceHost class.
+ *
+ * @property {string} ServiceHost.catalog - The catalog name.
+ * @property {string} ServiceHost.default - The default uri.
+ * @property {boolean} ServiceHost.failed - The failed status flag.
+ * @property {string} ServiceHost.hostGroup - The host group URI.
+ * @property {string} ServiceHost.id - The cluster ID.
+ * @property {number} ServiceHost.priority - The priority.
+ * @property {boolean} ServiceHost.replaced - The replaced status flag.
+ * @property {string} ServiceHost.uri - The URI.
+ */
+export default class ServiceHost {
+  /**
+   * The parameter transfer object for {@link constructor}.
+   *
+   * @typedef {object} ConstructorPTO
+   * @property {string} ConstructorPTO.catalog - The catalog.
+   * @property {string} ConstructorPTO.defaultUri - The default uri.
+   * @property {string} ConstructorPTO.hostGroup - The URI.
+   * @property {string} ConstructorPTO.id - The cluster ID.
+   * @property {number} ConstructorPTO.priority - The priority.
+   * @property {string} ConstructorPTO.uri - The URI.
+   */
+
+  /**
+   * Generate a new `ServiceHost`.
+   *
+   * @param {ConstructorPTO} pto - The parameter transfer object.
+   */
+  constructor(pto) {
+    // Validate the parameter transfer object.
+    ServiceHost.validate(pto);
+
+    // Map the parameter transfer object to the class object.
+    this.catalog = pto.catalog;
+    this.default = pto.defaultUri;
+    this.hostGroup = pto.hostGroup;
+    this.id = pto.id;
+    this.priority = pto.priority;
+    this.uri = pto.uri;
+
+    // Generate flags.
+    this.failed = false;
+    this.replaced = false;
+  }
+
+  /**
+   * Determine if the `ServiceHost` is in an active state.
+   *
+   * @returns {boolean} - `true` if the service is active and usable.
+   */
+  get active() {
+    // Validate that the `ServiceHost` was marked as failed.
+    if (this.failed) {
+      return false;
+    }
+
+    // Validate that the `ServiceHost` was replaced by another `ServiceHost`.
+    if (this.replaced) {
+      return false;
+    }
+
+    // Return true once all validation has passed.
+    return true;
+  }
+
+  /**
+   * Get the cluster value.
+   *
+   * @returns {string} - The cluster value.
+   */
+  get cluster() {
+    return this.id.split(':')[2];
+  }
+
+  /**
+   * Get the group value.
+   *
+   * @returns {string} - The group value.
+   */
+  get group() {
+    return this.id.split(':')[1];
+  }
+
+  /**
+   * Get the head value.
+   *
+   * @returns {string} - The head value.
+   */
+  get head() {
+    return this.id.split(':')[0];
+  }
+
+  /**
+   * Determine if the host is local to the user's cluster.
+   *
+   * @returns {boolean} - If the host is local.
+   */
+  get local() {
```

Comment:
This getter is used to determine if the host is local to the home cluster of the authenticated user.